vcd/resource_vcd_vapp_vm_test.go

```diff
@@ -73,12 +78,48 @@ func TestAccVcdVAppVm_Basic(t *testing.T) {
 				ImportStateIdFunc: importStateIdVappObject(testConfig, vappName2, vmName),
 				// These fields can't be retrieved from user data
 				ImportStateVerifyIgnore: []string{"template_name", "catalog_name", "network_name",
-					"initscript", "accept_all_eulas", "power_on", "computer_name"},
+					"initscript", "accept_all_eulas", "power_on", "computer_name", "override_template_disk"},
 			},
 		},
 	})
 }
 
+func testCheckInternalDiskNonStringOutputs(internalDiskSize int) resource.TestCheckFunc {
```

Comment:
is there any reason why we can't use `resource.TestCheckResourceAttr` without output? A lot less lines of code and easier to grasp what it is checking for in a list of checks.

Reply:
TestCheckResourceAttr works only with strings.

Reply:
It works with all values - https://github.com/terraform-providers/terraform-provider-vcd/blob/master/vcd/resource_vcd_edgegateway_test.go#L284

Reply:
it was my mistake, I check against outputs. TestCheckOutput don't work with anything other than strings

Reply:
Yes- but the whole point as I mentioned above  is that you have no extra gains from using `output` fields as you are not doing any type conversions. It just becomes ±8 lines instead of 1 and harder read

Reply:
Yes, I have. Internal disks are computed values and we don't have data sources for internal disks. I verify that computed values, not from configuration are accessible (e.g. outputus) and have same values as expected.

Reply:
But one can access indexes without any conversion.  We have done this before https://github.com/terraform-providers/terraform-provider-vcd/blob/master/vcd/resource_vcd_vapp_vm_multinetwork_test.go#L158

Instead of defining `output`  for all fields of and `func testCheckInternalDiskNonStringOutputs` one should be able to simply use one line check
`resource.TestCheckResourceAttr("vcd_vapp_vm.vmName", "internal_disk.0.size_in_mb", "20000"),`

Reply:
Again, I am saying that's it's the same as we use in other resources we test and we agree it has own benefits.

Reply:
But I am probably missing the point - what is the advantage?

Reply:
I'm pulling out this discussion from exact PR.